File: entity/StudentResponse.py
class StudentResponse:
    def __init__(self, student_id, subject_name, answers, total_score=0):
        self.student_id = student_id
        self.subject_name = subject_name
        self.answers = answers
        self.total_score = total_score

    def to_dict(self):
        # MongoDB에 저장할 수 있는 딕셔너리 형태로 변환
        return {
            "student_id": self.student_id,
            "subject_name": self.subject_name,
            "answers": self.answers,
            "total_score": self.total_score
        }

File: checkmate/BACKEND/src/main/java/com/checkmate/ai/entity/StudentResponse.java
package com.checkmate.ai.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)  // Auditing 활성화
public class StudentResponse {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long studentResponseId;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "student_id")
    private Student student;  // 연관관계 필드로 변경


    private String subject;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "student_response_id")
    private List<ExamResponse> answers = new ArrayList<>();

    private float totalScore;

    // 생성/수정일 자동 처리 (JPA Auditing 활성화 필요)
    @CreatedDate
    private LocalDateTime createdAt;
    @LastModifiedDate
    private LocalDateTime updatedAt;

}
